app/components/Wrapper.tsx

```diff
@@ -9,13 +10,58 @@ import RewardsHeader from "./RewardsHeader";
 export default function Wrapper({ children, network }: { children?: ReactElement; network: string }) {
   let link: string;
   let linkText: string;
-
+  
   const [isOpen, setIsOpen] = useState(false);
   const location = useLocation();
   const { address, connector } = useAccount();
   const { disconnect } = useDisconnect();
-  const { chain } = useNetwork();
-  const { switchNetwork } = useSwitchNetwork();
+  const { chain, chains } = useNetwork();
+  
+  const primaryChainObj = chains.filter(chainObj => {
+    return chainObj.name?.toLowerCase() === network
+  })[0]
+
+  const { switchNetwork } = useSwitchNetwork({ 
+    chainId: primaryChainObj?.id,
+    onError() {
+        tryRPCSwitchNetwork();
+    }
+  });
+
+  async function tryRPCSwitchNetwork() {
+    try {
+      console.log("Trying RPC switch chain");
+      await window?.ethereum?.request({
+        method: 'wallet_switchEthereumChain', 
+        params: [{ 
+          chainId: primaryChainObj?.id?.toString(16),
+        }]
+      })
+    }
+    catch (error) {
```

Comment:
the existing switch network functionality had an else statement messaging user to switch chain - I thought you could directly see by the switchNetwork functionality if that was supported - I feel like this isn't backwards compatible

Reply:
Hardhat network switching is officially beyond me. All other chains work fine except for hardhat. The popover for Metamask shows up and you can hit switch network, and the wagmi hook defines it as success but MM does not switch. I have raised issues with wagmi and Metamask to see if I can get an answer to this, but for now I have hidden the switch network button just for hardhat and brought back the "please switch to" message. Going to raise a new ticket for future work on this problem as well as the other wallet's nuances of changing networks.

Reply:
I'm still banking on the hardhat issue being a chainid issue 🙈 

While it sucks for developer experience, we can manually change the network in metamask to localhost and it'll connect-  I opened https://team-1612274056224.atlassian.net/browse/PD-94 to track that seperately